packages/web-client/app/components/card-pay/deposit-workflow/transaction-amount/index.hbs

```diff
@@ -1,60 +1,69 @@
       <Boxel::Field
-        @label="Amount to deposit"
-        @fieldMode="edit"
-        class="card-pay-transaction-amount__field"
-        @labelClass="card-pay-transaction-amount__field-label"
+        class="transaction-amount__field"
+        @label="Funding From:"
+        @labelClass="transaction-amount__from-field-label"
       >
-      <div class="card-pay-transaction-amount__amount-field-input-container">
-        {{svg-jar this.currentTokenDetails.icon class="card-pay-transaction-amount__amount-field-currency-icon" role="presentation"}}
-        {{!--
-          .boxel-input is a hack to get the boxel styles.
-          depending on how we plan to handle validation of this input,
-          we may need to change Boxel's input to use a more DDAU pattern
-        --}}
-        <input
-          class="boxel-input card-pay-transaction-amount__amount-field-input"
-          id="deposit-amount-input"
-          value={{this.amount}}
-          {{on "input" this.onInputAmount}}
-          data-test-deposit-amount-input
+        <CardPay::DepositWorkflow::TransactionAmount::BalanceView
+          @wallet={{this.layer1Network.chainName}}
+          @tokenSymbol={{this.currentTokenSymbol}}
+          @tokenIcon={{this.currentTokenDetails.icon}}
+          @address={{this.layer1Network.walletInfo.firstAddress}}
+          @balance={{format-token-amount this.currentTokenBalance}}
+          @balanceInUsd={{token-to-usd this.currentTokenSymbol this.currentTokenBalance}}
+          @isComplete={{@isComplete}}
         />
-        <div class="card-pay-transaction-amount__amount-field-currency-text">
-          {{this.currentTokenSymbol}}
-        </div>
-      </div>
       </Boxel::Field>
+      {{#if (or this.isUnlocking this.isUnlocked)}}
+        <Boxel::Field
+          class="transaction-amount__field"
+          @label="Amount to deposit:"
+        >
+          <div class="transaction-amount__token">
+            {{svg-jar this.currentTokenDetails.icon class="transaction-amount__token-icon" role="presentation" width="40" height="40"}}
+            <div>
+              <div class="transaction-amount__token-balance">
+                {{this.amount}} {{this.currentTokenSymbol}}
+              </div>
+              <div class="transaction-amount__usd-balance">
+                ≈ $ {{token-to-usd this.currentTokenSymbol this.amountAsBigNumber}} USD
+              </div>
+            </div>
+          </div>
+        </Boxel::Field>
+      {{else}}
+        <Boxel::Field
+          @label="Amount to deposit"
+          @fieldMode="edit"
+          class="transaction-amount__field"
+        >
+        <div class="transaction-amount__input-container transaction-amount__input-token-balance">
+          {{svg-jar this.currentTokenDetails.icon class="transaction-amount__currency-icon" role="presentation"}}
+          <Boxel::Input
+            class="transaction-amount__input"
+            id="deposit-amount-input"
+            value={{this.amount}}
```

Comment:
Not using `@value` with this input feels wrong, because Boxel's input uses Ember's `<Input>`, which says in its documentation that we should not use the html attribute.

https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/Input?anchor=Input

This does bypass 2 way binding, but I don't think we should be doing this. I was wondering if we should change Boxel's input not to use Ember's input but use the plain HTML input internally - we need more control than a magically working component gives us.

Reply:
+1 to using plain input rather than Ember's <Input> in most cases.